### PR TITLE
Get canonical version for publish URL

### PR DESCRIPTION
**Pull request by live1206**
[AB#230340](https://dev.azure.com/ceapex/Engineering/_workitems/edit/230340)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6053)

### CODE REVIEW

File: src/docfx/lib/markdown/validation/DocsValidationExtension.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ internal static class DocsValidationExtension
             MarkdownEngine markdownEngine,
             ContentValidator contentValidator,
             Func<MonikerList> getFileLevelMonikers,
-            Func<MonikerList> getPageLevelMonikers)
+            Func<string?> getPageLevelMonikers)
```
Comment (OsmondJiang) [inline src/docfx/lib/markdown/validation/DocsValidationExtension.cs:24]: >getPageLevelMonikers [](start = 26, length = 20)

rename

**Comment by OsmondJiang** [comment]
>                 var pageLevelMoniker = getPageLevelMonikers();

rename

---
Refers to: src/docfx/lib/markdown/validation/DocsValidationExtension.cs:36 in 671b9ac. [](commit_id = 671b9acf483e905fff60bb79ca68d795fdbf0da7, deletion_comment = False)

File: src/docfx/lib/markdown/MarkdownEngine.cs [hunk 2]
```diff
@@ -331,14 +338,9 @@ private MonikerList GetFileLevelMonikers()
...
+        private string? GetPageLevelMonikers()
         {
-            var status = t_status.Value!.Peek();
-
-            // todo: change to GetPageLevelMonikers when it's ready
-            var (monikerErrors, monikers) = _monikerProvider.GetFileLevelMonikers(((Document)InclusionContext.RootFile).FilePath);
-            status.Errors.AddRange(monikerErrors);
-            return monikers;
+            return _publishUrlMap!.GetCanonicalVersion(((Document)InclusionContext.RootFile).SiteUrl);
         }
```
Comment (OsmondJiang) [inline src/docfx/lib/markdown/MarkdownEngine.cs:344]: add some tests for split versioning cases?

File: src/docfx/lib/markdown/MarkdownEngine.cs [hunk 3]
```diff
@@ -74,6 +76,11 @@ internal class MarkdownEngine
             };
         }
 
+        public void Configure(PublishUrlMap publishUrlMap)
+        {
+            _publishUrlMap = publishUrlMap;
+        }
```
Comment (OsmondJiang) [inline src/docfx/lib/markdown/MarkdownEngine.cs:82]: 🤣🤣🤣🤣

File: src/docfx/build/moniker/MonikerProvider.cs [hunk 4]
```diff
@@ -182,5 +194,16 @@ private static (Error?, MonikerList) GetMonikerIntersection(UserMetadata metadat
...
             return default;
         }
+
+        private Dictionary<string, int> SetMonikerOrder(MonikerDefinitionModel monikerDefinition)
+        {
+            var result = new Dictionary<string, int>();
+            var monikers = monikerDefinition.Monikers;
+            for (var i = 0; i < monikers.Count; i++)
+            {
+                result[monikers[i].MonikerName] = i;
```
Comment (yufeih) [inline src/docfx/build/moniker/MonikerProvider.cs:204]: I don't see `Moniker.Order` property been used here. Seems like a bug to me.

**Pull request merged**